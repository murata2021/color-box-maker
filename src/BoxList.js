import React, { useState } from "react";
import Box from './Box'
import NewBoxForm from "./NewBoxForm";
import uuid from 'react-uuid';



const BoxList=()=>{

    const INITIAL_BOX_STATE=[]
    const [boxes,setBoxes]=useState(INITIAL_BOX_STATE)

    const addBox=(newBox)=>{
        newBox.id=uuid()
        setBoxes(boxes=>[...boxes,{...newBox,id:uuid()}])
    }

    const deleteBox=(id)=>{

        let copyBox=[...boxes.filter(box=>box.id!==id)]
        setBoxes(boxes=>copyBox)
    }

    return(
        <div className="container mt-3 ">
            <h2 className="d-flex justify-content-center">Let's make some color box ðŸ¥³ðŸ¥³ðŸ¥³</h2>
            <NewBoxForm addBox={addBox}/>
            <div className='row row-cols-2'>
                {boxes.map(box=><Box id={box.id} key={box.id} color={box.color} height={box.height} width={box.width} delete={deleteBox}/>)}
            </div>
        </div>
    )
}

export default BoxList;