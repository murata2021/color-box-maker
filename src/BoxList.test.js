import React from "react";
import {render,fireEvent} from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without crashing",()=>{
    render(<BoxList/>);
});

it("matches snapshot",()=>{
    const {asFragment}=render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});


it("adds new box",()=>{
    const {debug,queryByText,getByLabelText}=render(<BoxList/>);

    const btn=queryByText("Submit")
    let colorInput=getByLabelText('Color')
    let heightInput=getByLabelText('Height')
    let widthInput=getByLabelText('Width')

    
    expect(queryByText('Delete me!')).not.toBeInTheDocument()


    fireEvent.change(colorInput,{target:{value:'green'}})
    fireEvent.change(heightInput,{target:{value:'100px'}})
    fireEvent.change(widthInput,{target:{value:'150px'}})
    fireEvent.click(btn)

    let deleteMe=queryByText('Delete me!')
    
    expect(deleteMe.previousSibling).toHaveStyle(`
    width: 150px;
    height: 100px;
    background-color: green;
  `);

    //it clears form fields after submitting the form
    expect(colorInput.value).toBe("")
    expect(heightInput.value).toBe("")
    expect(widthInput.value).toBe("")



    expect(deleteMe).toBeInTheDocument()
    
});


it("deletes a box", function() {

    const {debug,queryByText,getByLabelText}=render(<BoxList/>);
    const btn=queryByText("Submit")
    let colorInput=getByLabelText('Color')
    let heightInput=getByLabelText('Height')
    let widthInput=getByLabelText('Width')

    expect(queryByText('Delete me!')).not.toBeInTheDocument()

    fireEvent.change(colorInput,{target:{value:'green'}})
    fireEvent.change(heightInput,{target:{value:'100px'}})
    fireEvent.change(widthInput,{target:{value:'150px'}})
    fireEvent.click(btn)

    let deleteMe=queryByText('Delete me!')

    fireEvent.click(deleteMe)

    expect(queryByText('Delete me!')).not.toBeInTheDocument()

  });