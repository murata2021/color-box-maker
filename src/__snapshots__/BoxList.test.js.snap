// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mt-3 "
  >
    <h2
      class="d-flex justify-content-center"
    >
      Let's make some color box ðŸ¥³ðŸ¥³ðŸ¥³
    </h2>
    <form>
      <div
        class="row g-3 align-items-center d-flex justify-content-center mt-4"
      >
        <div
          class="col-auto"
        >
          <label
            class="form-label"
            for="color"
          >
            Color
          </label>
          <input
            class="form-control"
            id="color"
            name="color"
            placeholder="color"
            type="text"
            value=""
          />
        </div>
        <div
          class="col-auto"
        >
          <label
            class="form-label"
            for="width"
          >
            Width
          </label>
          <input
            class="form-control"
            id="width"
            name="width"
            placeholder="width"
            type="text"
            value=""
          />
        </div>
        <div
          class="col-auto"
        >
          <label
            class="form-label"
            for="height"
          >
            Height
          </label>
          <input
            class="form-control"
            id="height"
            name="height"
            placeholder="height"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn btn-success mt-2"
          style="width: 50%;"
        >
          Submit
        </button>
      </div>
    </form>
    <div
      class="row row-cols-2"
    />
  </div>
</DocumentFragment>
`;
